packages/nouns-contracts/contracts/NounsAuctionHouseV2.sol

```diff
@@ -104,46 +104,44 @@ contract NounsAuctionHouseV2 is
      * @notice Settle the current auction.
      * @dev This function can only be called when the contract is paused.
      */
-    function settleAuction() external override whenPaused nonReentrant {
+    function settleAuction() external override whenPaused {
         _settleAuction();
     }
 
     /**
      * @notice Create a bid for a Noun, with a given amount.
      * @dev This contract only accepts payment in ETH.
      */
-    function createBid(uint256 nounId) external payable override nonReentrant {
-        INounsAuctionHouse.Auction memory _auction = auction;
-
-        require(_auction.nounId == nounId, 'Noun not up for auction');
-        require(block.timestamp < _auction.endTime, 'Auction expired');
-        require(msg.value >= reservePrice, 'Must send at least reservePrice');
-        require(
-            msg.value >= _auction.amount + ((_auction.amount * minBidIncrementPercentage) / 100),
-            'Must send more than last bid by minBidIncrementPercentage amount'
-        );
-
-        address payable lastBidder = _auction.bidder;
+    function createBid(uint256 nounId) external payable override {
+        INounsAuctionHouse.AuctionV2 memory _auction = auction;
 
-        // Refund the last bidder, if applicable
-        if (lastBidder != address(0)) {
-            _safeTransferETHWithFallback(lastBidder, _auction.amount);
-        }
+        if (_auction.nounId != nounId) revert NounNotUpForAuction();
+        if (block.timestamp >= _auction.endTime) revert AuctionExpired();
+        if (msg.value < reservePrice) revert MustSendAtLeastReservePrice();
+        if (msg.value < _auction.amount + ((_auction.amount * minBidIncrementPercentage) / 100))
+            revert BidDifferenceMustBeGreaterThanMinBidIncrement();
 
-        auction.amount = msg.value;
+        auction.amount = uint128(msg.value);
         auction.bidder = payable(msg.sender);
 
         // Extend the auction if the bid was received within `timeBuffer` of the auction end time
         bool extended = _auction.endTime - block.timestamp < timeBuffer;
         if (extended) {
-            auction.endTime = _auction.endTime = block.timestamp + timeBuffer;
+            auction.endTime = _auction.endTime = uint40(block.timestamp + timeBuffer);
         }
 
         emit AuctionBid(_auction.nounId, msg.sender, msg.value, extended);
 
         if (extended) {
             emit AuctionExtended(_auction.nounId, _auction.endTime);
         }
```

Comment:
Can reduce to a single if statement - 

```solidity
// Extend the auction if the bid was received within `timeBuffer` of the auction end time
bool extended = _auction.endTime - block.timestamp < timeBuffer;

emit AuctionBid(_auction.nounId, msg.sender, msg.value, extended);

if (extended) {
    auction.endTime = _auction.endTime = uint40(block.timestamp + timeBuffer);
    
    emit AuctionExtended(_auction.nounId, _auction.endTime);
}
```

Reply:
absolutely! I saw it and forgot to push it :)